cli/azd/cmd/container.go

```diff
@@ -557,6 +558,16 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 	container.MustRegisterNamedSingleton("ollama", llm.NewOllamaModelProvider)
 	container.MustRegisterNamedSingleton("azure", llm.NewAzureOpenAiModelProvider)
 
+	// Agent security manager
+	container.MustRegisterSingleton(func() (*security.Manager, error) {
```

Comment:
nit: This feels like an implementation detail of the agent factory, and could be moved there. 

`agent_factory.go`:

```golang
func (f *AgentFactory) Create(opts ...AgentCreateOption) (Agent, error) {
   securityManager := security.NewManager(cwd)
   ...
   localtools.NewLocalToolsLoader(securityManager),
}
```

Reply:
Technically we could move it there but I'd rather keep it where it is so that we continue to follow the same initialization strategy and use the `cwd` at time of `azd` initialization.